machine_learning/similarity_search.py

```diff
@@ -30,7 +31,9 @@ def euclidean(input_a: np.ndarray, input_b: np.ndarray) -> float:
     return math.sqrt(sum(pow(a - b, 2) for a, b in zip(input_a, input_b)))
 
 
-def similarity_search(dataset: np.ndarray, value_array: np.ndarray) -> list:
+def similarity_search(
+    dataset: np.ndarray, value_array: np.ndarray
+) -> List[List[float]]:
```

Comment:
The return type hint does not seem to be correct as per the given examples:
```
[[[0], 0.0]]
[[[0, 0, 0], 1.0]]
[[[0, 0, 0], 0.0], [[0, 0, 0], 1.0]]
```
Maybe: 
```python
List[List[Union[List[float], float]]]
```
I would suggest changing the return data type but that is out of scope for this PR.

Reply:
@dhruvmanila My mistake... you are right.
Have a question: I've understood return data will be List[List[List[float], float]].
Your type above means List[List[List[float]]] or List[List[float]], am I correct??

Reply:
Done :)